### ISSUE TITLE
Unit tests

### ISSUE THREAD

**Issue by JohnathonNow** [JohnathonNow/Bending#4]
Adding some unit tests would be nice, though the code is not at all built to support them.

**Comment by NyorJa** [comment]
Is it ok can I contribute an example for unit test?

**Comment by NyorJa** [comment]
Hi @JohnathonNow,

Just checked the code. I think mocking is impossible on classes. I am not sure what suitable framework would fit on this. Well, lets see

**Comment by JohnathonNow** [comment]
Yeah, I'll raise another issue to replace all direct class accesses with singletons or another appropriate pattern.

**Comment by b0n541** [comment]
Is this issue still open to be tackled?

**Comment by b0n541** [comment]
Please have a look whether my first changes go into the direction you had in mind for unit tests.

**Comment by JohnathonNow** [comment]
Those are what I had in mind, yes.
